react/components/AdvancedNotificationBar/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { memo, useState } from 'react'
+import { useCssHandles } from 'vtex.css-handles'
+import type { CssHandlesTypes } from 'vtex.css-handles'
+import { Link } from 'vtex.render-runtime'
+
+import AnnounceClose from './icons/AnnounceClose'
+import AnnounceRight from './icons/AnnounceRight'
+import AnnounceInfo from './icons/AnnounceInfo'
+
+const CSS_HANDLES = [
+  'notificationBarContainer',
+  'notificationBarInner',
+  'notificationContent',
+] as const
+
+interface Props {
+  content?: string
+  color?: string
+  link?: string
+  linkText?: string
+  icon?: string
+  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
+}
+
+function AdvancedNotificationBar({
+  content,
+  color,
+  link,
+  linkText,
+  icon,
+  classes,
+}: Props) {
+  const { handles } = useCssHandles(CSS_HANDLES, { classes })
+  const [show, setShow] = useState(true)
+
+  if (!content) {
+    return null
+  }
+
+  let background = ''
+  let iconBackground = ''
+  let secondaryTheme = false // for ex the yellow bar has brown icons and text
+  let fill = '#fff'
+
+  switch (color) {
+    case 'Dark Blue':
+      background = '#143D5F'
+      iconBackground = 'rgba(255, 255, 255, 0.1)'
+      break
+
+    case 'Green':
+      background = '#00AC6C'
+      iconBackground = 'rgba(0, 123, 77, 0.4)'
+      break
+
+    case 'Red':
+      background = '#ED002E'
+      iconBackground = '#D40029'
+      break
+
+    case 'Light Blue':
+      background = '#29A7CD'
+      iconBackground = '#1993B8'
+      break
+
+    case 'Yellow':
+      background = '#FFDE80'
+      iconBackground = 'rgba(204, 151, 0, 0.3)'
+      fill = '#775800'
+      secondaryTheme = true
+      break
+
+    default:
+      background = '#143D5F'
+      iconBackground = 'rgba(255, 255, 255, 0.1)'
+      break
+  }
+
+  const textColor = secondaryTheme ? '#775800' : '#fff'
+
+  return (
+    <>
+      {show && (
+        <div
```

Comment:
return show && <div

Reply:
Primesc erori de typescript, din cauza structurii (trebuie sa fie MemoExoticComponent, altfel da eroare in NotificationBar si nu pot schimba nimic acolo pt ca e default vtex, si return nu poate sa inceapa cu un if). Este neaparat sa schimbam aici?